lib/reference_model/reference_model.flist

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright 2024 Torje Nygaard Eikenes
+//
+// Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://solderpad.org/licenses/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+
+//+incdir+${RM_HOME}
+${RM_HOME}/iss_wrap_pkg.sv
+${RM_HOME}/pipeline_shell_stage.sv
+${RM_HOME}/pipeline_shell_controller_cv32e40s.sv
```

Comment:
With this, the common "reference_model" component has a hard dependency on a 40s-specific file?

I think that as many files as possible should be common, but the common code should not have a hard dependency.
Whether or not the core-specific file could live in a sort of "add-on"-like directory in the common dir or whether it should be in a core-specific dir (e.g. "tb/"), I am not sure.

Reply:
The idea is to encapsulate all the core-specific implementation details into the controller module, so only this would have to be modified for different cores. I am also unsure where it makes the most sense to place this file and how to choose between different cores. Since the controller is a module inside pipeline_shell, which is inside reference_model, it might be "ugly" to place this in "tb/"?

Reply:
Then there is just this and the other thread remaining.

> it might be "ugly" to place this in "tb/"?

If it logically can be made to fit there, that would be better than for other cores to be forced to use something 40s-specific.

Reply:
Moved the controller to the core-specific files now

Reply:
Looks good. Thanks.
(I will try it out with Xcelium and our typical flow. Will start new thread if anything goes wrong.)